lib/bolt/executor.rb

```diff
@@ -69,6 +77,9 @@ def on(nodes, callback = nil)
 
       @notifier.shutdown
 
+      # restore original handler
+      Signal.trap :INT, handler
```

Comment:
If you're trying to use bolt as a library, it seems odd to have signal handling here. I'd expect it to be handled by the application that is calling bolt (library) or the bolt cli itself.

Reply:
Yes, that's where I'm moving the signal handler registration.
I was originally convinced that registering the handler there has no effect, but that's not the case.